### DISCUSSION TITLE
virtio-fs: tag not found

### DISCUSSION THREAD

**Discussion by kvz** [utmapp/UTM#4703] [UTM Help (macOS)]
**Describe the issue**  
I'm not successful in mounting a directory on a virtualized Ubuntu ARM VM from a macOS 13 host using Apple Virtualization.

Steps completed inside the Ubuntu guest:

```bash
sudo apt install -y binfmt-support
sudo mkdir -p /media/rosetta
sudo mount -t virtiofs rosetta /media/rosetta

sudo /usr/sbin/update-binfmts --install rosetta /media/rosetta/rosetta \
  --magic "\x7fELF\x02\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x3e\x00" \
  --mask "\xff\xff\xff\xff\xff\xfe\xfe\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff" \
  --credentials yes --preserve no --fix-binary yes

grep -q ^rosetta /etc/fstab || echo "rosetta	/media/rosetta	virtiofs	ro,nofail	0	0" |sudo tee -a /etc/fstab

sudo modprobe virtiofs
sudo apt install spice-vdagent
```

And the VM is configured to have at least one `api2` tagged share as can be seen here:

<img width="1131" alt="image" src="https://user-images.githubusercontent.com/26752/203313643-1aa0d2ed-d72f-492c-bac0-380aab75eeb2.png">

I don't see the shares however in my `config.plist`, added below, is that to be expected?

Strangely enough, you can see virtiofs shared folders do work conceptually, because the Rosetta drive is mounted that way and I can confirm I can remount it without errors like so:

```bash
$ sudo umount -lf /media/rosetta
$ sudo mount -t virtiofs rosetta /media/rosetta
$ ls -al /media/rosetta
total 260
drwxr-xr-x 3 vagrant vagrant     96 Oct 26 09:27 .
-rwxr-xr-x 1 vagrant vagrant 540360 Sep 30 22:24 rosetta
```

So I think something goes wrong with the way that UTM allows me to share additional folders.

**Configuration**  
* UTM Version: 4.0.9
* macOS Version: 13.0.1 (22A400)
* Mac Chip: M1

**Crash log**  
```bash
$ sudo mkdir -p /srv/current
$ sudo chown -R ${USER} /srv/current
$ sudo mount -t virtiofs api2 /srv/current
mount: /srv/current: wrong fs type, bad option, bad superblock on api2, missing codepage or helper program, or other error.
```

**Debug log**  

```bash
$ sudo dmesg
[  735.874267] virtio-fs: tag <api2> not found
[  750.993736] virtio-fs: tag </api2> not found
[  756.242520] virtio-fs: tag </Users/kvz/code/api2> not found
```

As you can see I tried different variations of the `api2` shared folder tag, none of which worked.

**Upload VM**  
[config.plist.zip](https://github.com/utmapp/UTM/files/10067320/config.plist.zip)

**Comment by osy** [comment]
Please see https://docs.getutm.app/guest-support/linux/#virtfs the correct command is

```
$ sudo mount -t 9p -o trans=virtio share [mount point] -oversion=9p2000.L
```

**Reply by kvz** [reply]
I'm using Apple Virtualization, and for that the docs indicate that it really should be `sudo mount -t virtiofs api2 /srv/current`. I did try the 9p command before but it says no channel available. But yeah the docs seem pretty clear that for Apple Virtualization you should not 9p. Also when I look at how the rosette mount is contructed, it seems to be in the same manner (`virtiofs`).

![image](https://user-images.githubusercontent.com/26752/203520428-c3220b32-d859-4f21-9542-b059bd10082a.png)

**Reply by osy** [reply]
No the tag is still “share”

**Reply by kvz** [reply]
Amazing, that did the trick. I was very much thinking that I would have one tag for each share. But now I understand that `share` is hardcoded, and it contains each share inside of that. I think I should be able to mount bind them to different places, is that your recommendation also? (in case you need the shared folders at entirely different paths and names than they are presented in the bundle)

**Reply by kvz** [reply]
I got it working with mount binds at any rate, so thank you very much @osy 🎉 

Here are my full instructions for mounting two shared folders nested inside one another on Ubuntu ARM 22 with Apple Virtualization on an M1 in case someone else stumbles into something similar:

```bash
sudo grep -q ^virtiofs /etc/initramfs-tools/modules || echo "virtiofs" |sudo tee -a /etc/initramfs-tools/modules 
sudo update-initramfs -uk all
sudo modprobe virtiofs 
sudo mkdir -p /media/share
sudo umount -lf /media/share
sudo mkdir -p /srv/current/api2
sudo grep -q ^share /etc/fstab || echo "share	/media/share	virtiofs	rw,nofail	0	0" |sudo tee -a /etc/fstab
sudo grep -q ^/media/share/api2 /etc/fstab || echo "/media/share/api2	/srv/current	none	defaults,bind	0	0" |sudo tee -a /etc/fstab
sudo grep -q ^/media/share/content /etc/fstab || echo "/media/share/content	/srv/current/crm/webroot	none	defaults,bind	0	0" |sudo tee -a /etc/fstab
sudo chown -R ${USER} /media/share
sudo mount -a
# sudo journalctl -xe
```

**Reply by saoj** [reply]
Would you mind sharing how you solved this? When I try to mount on Ventura I get the following:
```
$ sudo mount -t virtiofs vms_shared_folder /Users/soliveira/vms_shared_folder
mount_virtiofs: fs_tag vms_shared_folder not found
mount: /Users/soliveira/vms_shared_folder failed with 1
```

**Reply by osy** [reply]
https://docs.getutm.app/guest-support/macos/#virtiofs

**Reply by saoj** [reply]
My bad. This needs to be executed in the virtual machine (Ubuntu), not in the host machine (Mac). And the correct command for the virtual machine (Ubuntu) is:

    sudo mount -t 9p -o trans=virtio share /home/soliveira/host-wp -oversion=9p2000.L

That's of course after you have properly configured the sharing folder under UTM configurations for your VM.

**Reply by kokoskiwi** [reply]
please help me @osy @kvz @saoj  https://github.com/utmapp/UTM/issues/6078

**Comment by burningalchemist** [comment]
Hey @osy, I bumped into the same issue on `MacOS 14.1` (both host/guest) after the update:

```
~ sw_vers
ProductName:            macOS
ProductVersion:         14.1
BuildVersion:           23B74
~ mkdir -m 777 -p hostshare
~ file hostshare
hostshare: directory
~ mount_virtiofs share hostshare
mount_virtiofs: fs_tag share not found
~
```

It worked well for me since 13+. I'm wondering if I'm missing something or it's a common (new) issue. Tried to restart the host, the vm, re-adding the shared directory with no success.

UTM Version is 4.4.4, AVF.

Thank you. :)

**Reply by burningalchemist** [reply]
Seems to be a solution: https://github.com/utmapp/UTM/issues/5875#issuecomment-1806957803 👍

**Comment by nkinar** [comment]
After the initial install of UTM, the **My Shared Files** location did not show up in the Finder until a reboot of the guest.  Then, the **My Shared Files** location showed up under Locations in the Finder on the left sidebar.
